### DISCUSSION TITLE
How do I create a character save file (plr) for a server side character?

### DISCUSSION THREAD

**Discussion by Dobbelklick** [Pryaxis/TShock#3042] [Q&A] [answered]
Something like /downloadssc or /dumpssc (create serverside plr files) would be nice. I have full control over the server, but I really need a local copy of my character. How do I do that?

**Comment by sgkoishi** [comment, marked as answer]
You need a plugin like
* [this](https://github.com/sgkoishi/yaaiomni) (disclaimer: I wrote this) 
* or [this which is a Chinese plugin](https://github.com/hufang360/TShockPlayerManager) 
* or [this which is also Chinese](https://github.com/Megghy/PlayerExport)
* or [this which only exports online players](https://github.com/Qyriad/tshock-exportscc/)

**Comment by Dobbelklick** [comment]
THANK YOU!

Unfortunately I couldn't get your plugin to work, but the second one (PlayerManager) worked flawlessly.

Yours gave me this error message:

TShock encountered a problem from which it cannot recover. The following message may help diagnose the problem.
Until the problem is resolved, TShock will not be able to start (and will crash on startup).
System.ArgumentException: An item with the same key has already been added. Key: -ip
   at System.Collections.Generic.Dictionary`2.TryInsert(TKey key, TValue value, InsertionBehavior behavior)
   at System.Collections.Generic.Dictionary`2.Add(TKey key, TValue value)
   at Chireiden.TShock.Omni.Plugin.PostTShockInitialize() in /_/Core/Plugin.cs:line 424
   at TShockAPI.TShock.Initialize()

I removed all other plugins beforehand.

**Reply by sgkoishi** [reply]
Thanks for the feedback! (it seems that you set a rarely used `-ip` cli parameter)

**Reply by Dobbelklick** [reply]
Yeah, I'm using one of the game eggs (a kind of preconfigured server image) of [Pterodactyl](https://pterodactyl.io/) and the default startup command is `./TShock.Server -ip 0.0.0.0 -port 7777 -maxplayers 8 -world world.wld -autocreate 1`.
